scripts/cluster/join.py

```diff
@@ -36,6 +38,23 @@
 cluster_key_file = "{}/cluster.key".format(cluster_dir)
 
 
+def get_fingerprint(addr, port):
+    """
+    Get the certificate fingerprint of the server at addr:port
+
+    :param addr: the address of the server
+    :param port: the port of the server
+    :return: the digest of the server cert
+    """
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.settimeout(1)
+    wrapped_socket = ssl.wrap_socket(sock)
+    wrapped_socket.connect((addr, port))
+    der_cert_bin = wrapped_socket.getpeercert(True)
+    wrapped_socket.close()
+    return hashlib.sha256(der_cert_bin).hexdigest()
```

Comment:
If the peer doesn't supply a certificate, this step will fail with exception `TypeError: object supporting the buffer API required` -- please make sure this exception either terminates the application or otherwise prevents the connection to untrusted sources.

Reply:
If anything goes wrong with the`get_fingerprint` we catch the top level `Exception` and exit gracefully, see https://github.com/ubuntu/microk8s/pull/1263/files#diff-7ffd8f66475be19b5114657ee59733edR789 . I believe this is the behavior you expect.